[COCOA] Add support for web-platform-tests custom domains when running layout tests

cdumez opened the pull request:
#### 4f0d22ad0693fcbd1ff376ffb1809c1f79914ef2
<pre>
[COCOA] Add support for web-platform-tests custom domains when running layout tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=245294">https://bugs.webkit.org/show_bug.cgi?id=245294</a>

Reviewed by NOBODY (OOPS!).

Leverage CFNetwork SPI to support web-platform-tests custom domains when running
WPT tests on Cocoa ports. Glib ports already support this.

* LayoutTests/imported/w3c/web-platform-tests/*:
Rebaseline WPT tests now that they are properly run on the web-platform-tests custom
domain instead of localhost. Some more tests can be unskipped too but I&apos;ll do this
in a follow-up patch given the patch size.
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4f0d22ad0693fcbd1ff376ffb1809c1f79914ef2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/92037 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/1276 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/22565 "Built successfully") | [❌ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/101840 "Hash 4f0d22ad for PR 4438 does not build (failure)") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/161906 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/1270 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/29690 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/84493 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/98029 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/97695 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/785 "Found 30 new test failures: compositing/reflections/mask-and-reflection.html, editing/execCommand/insert-ordered-list-and-delete.html, editing/selection/ios/change-selection-by-tapping.html, fast/canvas/image-pattern-rotate.html, http/tests/app-privacy-report/app-attribution-beacon-isappinitiated.html, http/tests/app-privacy-report/app-attribution-beacon-isnotappinitiated.html, http/tests/app-privacy-report/user-attribution-cors-preflight-redirect.html, http/tests/blink/sendbeacon/beacon-cookie.html, http/tests/workers/service/serviceworker-websocket.https.html, http/wpt/beacon/cors/crossorigin-arraybufferview-no-preflight.html ... (failure)") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/78576 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/27737 "Passed tests") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/82707 "Found 11 new API test failures: TestWebKitAPI.WKDownload.ChallengeFailure, TestWebKitAPI.TLSVersion.ShouldAllowDeprecatedTLS, TestWebKitAPI.Preconnect.HTTPS, TestWebKitAPI.TLSVersion.NetworkSessionNSUserDefaults, TestWebKitAPI.RemoteObjectRegistry.SerializeErrorWithCertificates, TestWebKitAPI.WebKit.FastServerTrust, TestWebKitAPI.WKDownload.ChallengeSuccess, TestWebKitAPI.WebKit.ServerTrust, TestWebKitAPI.TLSVersion.DidNegotiateModernTLS, TestWebKitAPI.Challenge.ClientCertificate ... (failure)") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/82317 "Found 11 new API test failures: TestWebKitAPI.WKDownload.ChallengeFailure, TestWebKitAPI.TLSVersion.ShouldAllowDeprecatedTLS, TestWebKitAPI.Preconnect.HTTPS, TestWebKitAPI.TLSVersion.NetworkSessionNSUserDefaults, TestWebKitAPI.RemoteObjectRegistry.SerializeErrorWithCertificates, TestWebKitAPI.WebKit.FastServerTrust, TestWebKitAPI.WKDownload.ChallengeSuccess, TestWebKitAPI.WebKit.ServerTrust, TestWebKitAPI.TLSVersion.DidNegotiateModernTLS, TestWebKitAPI.Challenge.ClientCertificate ... (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/70775 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/36113 "Built successfully") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/16333 "Found 30 new test failures: http/tests/appcache/fail-on-update-2.html, http/tests/appcache/remove-cache.html, imported/w3c/web-platform-tests/IndexedDB/structured-clone.any.worker.html?101-last, imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/script-src-wildcard/script-tag.http.html, imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/script-src-wildcard/script-tag.https.html, imported/w3c/web-platform-tests/content-security-policy/gen/top.meta/script-src-wildcard/script-tag.http.html, imported/w3c/web-platform-tests/content-security-policy/gen/top.meta/script-src-wildcard/script-tag.https.html, imported/w3c/web-platform-tests/content-security-policy/generic/generic-0_10.sub.html, imported/w3c/web-platform-tests/content-security-policy/img-src/img-src-4_1.sub.html, imported/w3c/web-platform-tests/content-security-policy/img-src/img-src-host-partial-wildcard-allowed.sub.html ... (failure)") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/33856 "Built successfully") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/17435 "Found 30 new test failures: http/tests/cache/disk-cache/resource-becomes-uncacheable.html, http/wpt/beacon/cors/crossorigin-arraybufferview-no-preflight.html, http/wpt/cache-storage/cache-quota.any.html, http/wpt/fetch/response-opaque-clone.html, http/wpt/service-workers/server-trust-evaluation.https.html, imported/w3c/web-platform-tests/beacon/beacon-cors.https.window.html, imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/script-src-wildcard/script-tag.http.html, imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/script-src-wildcard/script-tag.https.html, imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/script-src-wildcard/sharedworker-import.https.html, imported/w3c/web-platform-tests/content-security-policy/gen/top.meta/script-src-wildcard/script-tag.http.html ... (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/37731 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/40145 "Found 30 new test failures: http/wpt/beacon/cors/crossorigin-arraybufferview-no-preflight.html, http/wpt/service-workers/server-trust-evaluation.https.html, imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/script-src-wildcard/script-tag.http.html, imported/w3c/web-platform-tests/content-security-policy/gen/top.http-rp/script-src-wildcard/script-tag.https.html, imported/w3c/web-platform-tests/content-security-policy/gen/top.meta/script-src-wildcard/script-tag.http.html, imported/w3c/web-platform-tests/content-security-policy/gen/top.meta/script-src-wildcard/script-tag.https.html, imported/w3c/web-platform-tests/content-security-policy/generic/generic-0_10.sub.html, imported/w3c/web-platform-tests/content-security-policy/img-src/img-src-4_1.sub.html, imported/w3c/web-platform-tests/content-security-policy/img-src/img-src-host-partial-wildcard-allowed.sub.html, imported/w3c/web-platform-tests/content-security-policy/img-src/img-src-port-wildcard-allowed.sub.html ... (failure)") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/39617 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/36561 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

TingPing reviewed (commented):
Not familiar with cocoa API but overall this looks excellent, really happy this was doable for the Apple ports!

cdumez committed:
[COCOA] Add support for web-platform-tests custom domains when running layout tests
https://bugs.webkit.org/show_bug.cgi?id=245294

Reviewed by NOBODY (OOPS!).

Leverage CFNetwork SPI to support web-platform-tests custom domains when running
WPT tests on Cocoa ports. Glib ports already support this.

* LayoutTests/imported/w3c/web-platform-tests/*:
Rebaseline WPT tests now that they are properly run on the web-platform-tests custom
domain instead of localhost. Some more tests can be unskipped too but I'll do this
in a follow-up patch given the patch size.
